src/backend/Csrs.Api/Features/UserRequests/Create.cs

```diff
@@ -1,27 +1,126 @@
+            {
+                FileNo = fileNo;
+                RequestType = requestType;
+                RequestMessage = requestMessage;
+            }
+            public string FileNo { get; init; }
+            public string RequestType { get; init; }
+            public string RequestMessage { get; init; }
         }
         public class Response
         {
+            public Response(string _response)
+            {
+                this._response = _response;
+            }
+
+            public string _response { get; init; }
         }
         public class Handler : IRequestHandler<Request, Response>
         {
+            private readonly IDynamicsClient _dynamicsClient;
+            private readonly IUserService _userService;
+            private readonly IAccountService _accountService;
+            private readonly IFileService _fileService;
             private readonly ILogger<Handler> _logger;
 
-            public Handler(ILogger<Handler> logger)
+            public Handler(
+                IDynamicsClient dynamicsClient,
+                IUserService userService,
+                IAccountService accountService,
+                IFileService fileService,
+                ILogger<Handler> logger)
             {
+                _dynamicsClient = dynamicsClient ?? throw new ArgumentNullException(nameof(dynamicsClient));
+                _userService = userService ?? throw new ArgumentNullException(nameof(userService));
+                _accountService = accountService ?? throw new ArgumentNullException(nameof(accountService));
+                _fileService = fileService ?? throw new ArgumentNullException(nameof(fileService));
                 _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             }
 
-            public Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
             {
-                return Task.FromResult(new Response());
+                _logger.LogDebug("Checking current user for BCeID Guid");
+
+                string userId = _userService.GetBCeIDUserId();
+                //string userId = "44e88bfd-fed0-4a68-97f7-0f701df51315";
+                if (string.IsNullOrEmpty(userId))
+                {
+                    _logger.LogInformation("No BCeID on authenticated user, cannot create account");
+                    return new Response("Unauthenticated");
+                }
+                var bceidScope = _logger.AddBCeIdGuid(userId);
+                // find to see if the person has an account already?
+                MicrosoftDynamicsCRMssgCsrspartyCollection parties = await _dynamicsClient.GetPartyByBCeIdAsync(userId, cancellationToken);
+                MicrosoftDynamicsCRMssgCsrsparty party;
+                if (parties.Value.Count > 0)
+                {
+                    party = parties.Value.First();
```

Comment:
You can just use `[0]` instead of using LINQ. But again, not critical.

Reply:
Noted.  maybe we should have a method to get partybyID that only returns 1 (similar to how I did byFileNo so the calling class dont have to work.   I will look at that in future PR